### PR TITLE
Tested #23622 -- ordering retained for distinct on fields subqueries

### PR DESCRIPTION
**Pull request by akaariai**
The ticket was already fixed by
b68212f539f206679580afbfd008e7d329c9cd31.

Thanks to Beauhurst for commissioning the work on this ticket.

### CODE REVIEW

File: tests/queries/tests.py [hunk 1]
```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
...
+
+
+class Ticket23622Tests(TestCase):
+    @skipUnlessDBFeature('can_distinct_on_fields')
+    def test_ticket_23622(self):
+        a1 = Ticket23605A()
+        a1.save()
+        a2 = Ticket23605A()
+        a2.save()
+        c1 = Ticket23605C.objects.create(field_c0=0.0)
```
Comment (timgraham) [inline tests/queries/tests.py]: Is there a reason to mix `create()` and `save()`?

File: tests/queries/tests.py [hunk 2]
```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
...
+        Ticket23605B(
+            modela_fk=a1, field_b0=123,
+            field_b1=datetime.date(2013, 1, 6),
+            modelc_fk=c1
```
Comment (timgraham) [inline tests/queries/tests.py]: add trailing comma on kwargs

File: tests/queries/tests.py [hunk 3]
```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
...
+        Ticket23605B(
+            modela_fk=a2, field_b0=56,
+            field_b1=datetime.date(2011, 1, 27),
+            modelc_fk=c1
+        ).save()
+
+        # Make sure __pk__in and __in work the same for related fields when
+        # using a distinct on subquery.
+        qx = (
+            Q(ticket23605b__pk__in=Ticket23605B.objects.order_by("modela_fk", "-field_b1").distinct("modela_fk"))
```
Comment (timgraham) [inline tests/queries/tests.py]: use single quotes

File: tests/queries/tests.py [hunk 4]
```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
...
+            & Q(ticket23605b__field_b0__gte=50)
+        )
+
+        qy = (
+            Q(ticket23605b__in=Ticket23605B.objects.order_by("modela_fk", "-field_b1").distinct("modela_fk"))
+            & Q(ticket23605b__field_b0__gte=50)
+        )
+        self.assertEqual(
```
Comment (timgraham) [inline tests/queries/tests.py]: do you want to add an assertion for the results of the queryset to assure that doesn't change?

File: tests/queries/tests.py [hunk 5]
```diff
@@ -3725,3 +3725,69 @@ def test_ticket_24605(self):
...
+            field_b1=datetime.date(2012, 10, 20),
+            modelc_fk=c1
+        ).save()
+        Ticket23605B(
+            modela_fk=a2, field_b0=56,
+            field_b1=datetime.date(2011, 1, 27),
+            modelc_fk=c1
+        ).save()
+
+        # Make sure __pk__in and __in work the same for related fields when
```
Comment (timgraham) [inline tests/queries/tests.py]: this could be the docstring so it's more clear upfront what the purpose of the test is?

**Comment by akaariai** [comment]
I fixed the review comments. I also fixed a test that had assumptions about the pk values in separate commit.

I quickly checked both PEP-8 and our style guide and didn't see where we recommend trailing commas for kwargs or use of single quote characters vs double quote characters. If we have such rules, fine, but if not I'm not that enthusiastic about adding such rules. Especially if we can't automatically check them.

**Comment by timgraham** [comment]
merged in 6a9a9e50f233b3ba4af13bf2dc664dfa3b12038d and baa732ac9f18b6ed0ca386c632e15c9fb3037e39, thanks!

**Pull request closed without merging**